components/constellation/constellation.rs

```diff
@@ -2092,6 +2102,13 @@ where
             }
         }
 
+        if let Some(webgpu) = self.webgpu.as_ref() {
+            debug!("Exiting WebGPU thread.");
+            if let Err(e) = webgpu.exit() {
```

Comment:
I would say that inside `exit`, you wan to add a `IpcSender<()>` to the exit message, and then block on the reply.

Because with the current setup, you can know for sure that you've queued the exit message, however there might be any number of other messages in front of it, therefore webgpu could keep running for an undetermined amount of time after the constellation has already exited.

Reply:
Addressed this in my latest commit: https://github.com/servo/servo/pull/24708/commits/40f37812f9789334a10a82a56460f9726dfdee95